### PR TITLE
[SPARK-39277][SQL] Make Optimizer extends SQLConfHelper and remove incorrect comment 

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

Make Optimizer extends SQLConfHelper and remove incorrect comment.

### Why are the changes needed?

Make the code easier to maintain.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing test.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala [hunk 1]
```diff
@@ -40,7 +41,7 @@ import org.apache.spark.util.Utils
  * Optimizers can override this.
  */
 abstract class Optimizer(catalogManager: CatalogManager)
-  extends RuleExecutor[LogicalPlan] {
+  extends RuleExecutor[LogicalPlan] with SQLConfHelper {
```
Comment (wangyum) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala:44]: Analyzer also extends SQLConfHelper.
https://github.com/apache/spark/blob/efc1e8ac8bc61872601ac2244629a9d54f8889fb/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L188-L189

**Comment by wangyum** [comment]
cc @cloud-fan

**Comment by wangyum** [comment]
cc @cloud-fan

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**